truffle/src/main/ruby/core/type.rb

```diff
@@ -82,6 +82,16 @@ def self.object_respond_to?(obj, name, include_private = false)
       Truffle.invoke_primitive :vm_object_respond_to, obj, name, include_private
     end
 
+    def self.object_respond_to_no_built_in?(obj, name, include_private = false)
+      meth = Truffle.invoke_primitive :vm_method_lookup, obj, name
+      !meth.nil? && !(Truffle.invoke_primitive :vm_method_is_basic, meth)
```

Comment:
I would move parenthesis around the call here: `!Truffle.invoke_primitive(:vm_method_is_basic, meth)`